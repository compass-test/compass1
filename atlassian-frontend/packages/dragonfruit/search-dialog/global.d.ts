/**
 * Strictly for test and storybook global type definition only, be careful with adding types here as it can theoretically
 * propagate to production files too.
 */

import 'jest-extended';
