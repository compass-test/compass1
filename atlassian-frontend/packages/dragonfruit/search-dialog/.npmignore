build/
docs/
src/
test/
examples/
tsconfig.json
analytics.spec.yaml
