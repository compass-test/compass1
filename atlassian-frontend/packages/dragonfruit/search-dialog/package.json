{
  "name": "@atlassian/compass-search-dialog",
  "version": "0.26.5",
  "description": "Integrated search dialog for Confluence and Jira",
  "publishConfig": {
    "registry": "https://packages.atlassian.com/api/npm/npm-remote"
  },
  "repository": "https://bitbucket.org/atlassian/atlassian-frontend",
  "author": "Atlassian Pty Ltd",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "module:es2019": "dist/es2019/index.js",
  "types": "dist/types/index.d.ts",
  "sideEffects": false,
  "atlaskit:src": "src/index.tsx",
  "atlassian": {
    "team": "Compass Lodestone",
    "releaseModel": "continuous"
  },
  "dependencies": {
    "@atlaskit/analytics-gas-types": "^5.0.0",
    "@atlaskit/analytics-next": "^8.1.0",
    "@atlaskit/atlassian-navigation": "^1.1.0",
    "@atlaskit/avatar": "^20.5.0",
    "@atlaskit/button": "^16.0.0",
    "@atlaskit/feedback-collector": "^7.1.0",
    "@atlaskit/flag": "^14.4.0",
    "@atlaskit/icon": "^21.9.0",
    "@atlaskit/icon-file-type": "^6.3.0",
    "@atlaskit/icon-object": "^6.2.0",
    "@atlaskit/section-message": "^6.0.0",
    "@atlaskit/spinner": "^15.0.0",
    "@atlaskit/tabs": "^13.1.0",
    "@atlaskit/theme": "^12.0.0",
    "@atlaskit/user-picker": "^7.8.0",
    "@atlaskit/util-service-support": "^6.0.0",
    "@atlassian/analytics-bridge": "^0.4.0",
    "@atlassian/compass-search-cache": "^0.6.0",
    "@atlassian/dragonfruit-components": "^0.22.0",
    "@atlassian/dragonfruit-graphql": "^3.21.0",
    "@atlassian/dragonfruit-rest": "^0.3.0",
    "@atlassian/dragonfruit-routes": "^1.2.0",
    "@atlassian/dragonfruit-teams": "^2.16.0",
    "@atlassian/dragonfruit-tenant-context": "^2.0.0",
    "@atlassian/dragonfruit-utils": "^1.7.0",
    "@atlassian/search-dialog": "^7.4.0",
    "@atlassiansox/engagekit": "^11.1.0",
    "@babel/runtime": "^7.0.0",
    "@emotion/core": "^10.0.9",
    "@emotion/styled": "^10.0.7",
    "debounce-promise": "^3.1.2",
    "deep-equal": "^1.0.1",
    "lodash": "^4.17.15",
    "memoize-one": "^6.0.0",
    "react-async-component": "^2.0.0",
    "rusha": "^0.8.13",
    "uuid": "^3.1.0"
  },
  "peerDependencies": {
    "jest": "^26.4.2",
    "jest-extended": "^0.11.2",
    "jest-when": "^2.7.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0",
    "react-intl": "^2.6.0",
    "react-select": "^4.3.1",
    "styled-components": "^3.2.6"
  },
  "devDependencies": {
    "@atlaskit/badge": "^15.0.0",
    "@atlaskit/docs": "*",
    "@atlaskit/drawer": "^7.0.0",
    "@atlaskit/logo": "^13.3.0",
    "@atlaskit/popup": "^1.1.0",
    "@atlassian/atlassian-frontend-prettier-config-1.0.1": "npm:@atlassian/atlassian-frontend-prettier-config@1.0.1",
    "@atlassian/dragonfruit-feature-flags": "^1.25.0",
    "@atlassian/dragonfruit-testing": "^1.0.12",
    "@storybook/addon-actions": "^5.2.5",
    "@storybook/addon-knobs": "^5.3.18",
    "@testing-library/dom": "^7.7.3",
    "@testing-library/react": "^8.0.1",
    "@types/faker": "^4.1.8",
    "@types/jest-when": "^2.7.1",
    "enzyme": "^3.10.0",
    "faker": "^4.1.0",
    "fetch-mock": "^8.0.0",
    "react-intl": "^2.6.0"
  },
  "homepage": "https://bitbucket.org/atlassian/atlassian-frontend/src/master/packages/search/product-search-dialog/#readme",
  "af:exports": {
    "./skeleton": "./src/skeleton.tsx",
    "./search-client": "./src/search-client.tsx",
    ".": "./src/index.tsx"
  },
  "engines": {
    "node": ">= 10.4.2"
  },
  "prettier": "@atlassian/atlassian-frontend-prettier-config-1.0.1"
}
