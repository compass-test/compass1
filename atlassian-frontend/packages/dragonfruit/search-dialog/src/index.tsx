import React from 'react';
import ProductSearchDialog from './cross-product/async-loading-wrapper';
import { Products } from './common/product-context';
import { CrossProductSearchDialogProps } from './cross-product';

// Confluence
import { ConfluenceFeaturesOverrides } from './confluence/confluence-features';

export {
  ConfluenceClientsProvider,
  ConfluenceSearchClient,
  Scope,
  ConfluencePage,
  ConfluenceBlogpost,
  ConfluenceAttachment,
  ConfluenceObjectResult,
  RecentConfluence,
} from './confluence/clients';

export type {
  ConfluenceClientsProviderProps,
  ConfSpaceResults,
  ConfItemResults,
  ConfPeopleResults,
  ContentType as ConfluenceContentType,
} from './confluence/clients';

// Common
export type { Site } from './common/clients';

export { ProductSearchInputSkeleton } from './common/product-search-input-skeleton';

export { ProductSearchDialog, Products };

// Deprecating old API
export const CrossProductSearchDialog = ProductSearchDialog;

export type { CrossProductSearchDialogProps } from './cross-product';

type ConfluenceSearchDialogProps = Omit<
  CrossProductSearchDialogProps,
  'confluenceFeatures' | 'products'
> & {
  features: ConfluenceFeaturesOverrides;
  forwardRef?: React.Ref<HTMLInputElement>;
};

export const ConfluenceSearchDialog = ({
  features,
  ...props
}: ConfluenceSearchDialogProps) => (
  <ProductSearchDialog {...props} products={[Products.compass]} />
);

export type { UserDetails } from './common/user-context';
