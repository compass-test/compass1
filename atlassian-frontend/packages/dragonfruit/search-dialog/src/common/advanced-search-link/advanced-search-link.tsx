import React, { FunctionComponent } from 'react';
import { LinkComponent } from '@atlassian/search-dialog';
import { StyledAdvancedSearch } from './advanced-search-link.styled';

export interface Props {
  href: string;
  linkComponent?: LinkComponent;
  isKeyboardHighlighted: boolean;
  onClick?: (e: React.MouseEvent | KeyboardEvent) => any;
}

export const AdvancedSearchLink: FunctionComponent<Props> = ({
  href,
  linkComponent,
  ...rest
}) => (
  <StyledAdvancedSearch
    linkComponent={linkComponent}
    data-test-id="search-dialog-advanced-search-link"
    href={href}
    {...rest}
  />
);
