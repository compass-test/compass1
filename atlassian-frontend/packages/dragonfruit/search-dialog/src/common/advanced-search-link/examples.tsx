import React from 'react';
import { SearchDialog, SearchFooter } from '@atlassian/search-dialog';
import { action } from '@storybook/addon-actions';
import { InjectedIntlProps, IntlProvider } from 'react-intl';
import { Products } from '../../common/product-context';
import { generateMockSitesWithAdvancedSearch } from '../../utils/site-util';
import {
  MultiSiteAdvancedSearchLink,
  Props,
} from './multi-site-advanced-search-link';

const commonProps: Props = {
  sites: [...generateMockSitesWithAdvancedSearch(5, Products.confluence)],
  isLoading: false,
  onClick: (href, e) => {
    e.preventDefault();
    return action(`onclick ${href}`)(e);
  },
};

export const MultiSite = (props: InjectedIntlProps) => (
  <MultiSiteAdvancedSearchLink {...props} {...commonProps} />
);

export const ManySites = (props: InjectedIntlProps) => (
  <MultiSiteAdvancedSearchLink
    {...props}
    {...commonProps}
    sites={generateMockSitesWithAdvancedSearch(100, Products.confluence)}
  />
);

export const NoDropdown = (props: InjectedIntlProps) => (
  <MultiSiteAdvancedSearchLink
    {...props}
    {...commonProps}
    sites={generateMockSitesWithAdvancedSearch(4, Products.confluence)}
  />
);

export default {
  title: 'common/Advanced Search Link',
  decorators: [
    (story: () => React.ElementType) => (
      <IntlProvider locale="en">
        <SearchDialog>
          <SearchFooter>{story()}</SearchFooter>
        </SearchDialog>
      </IntlProvider>
    ),
  ],
};
