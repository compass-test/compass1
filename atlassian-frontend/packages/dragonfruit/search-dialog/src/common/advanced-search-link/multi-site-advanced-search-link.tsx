import { LinkComponent, SearchFooterLinks } from '@atlassian/search-dialog';
import React, { FunctionComponent, useCallback } from 'react';
import { InjectedIntlProps, injectIntl } from 'react-intl';
import {
  getTrigger,
  isNewTab,
  onMultiSiteAdvancedSearchSelected,
  useAnalytics,
} from '../analytics';
import { Site } from '../clients';
import { Products } from '../product-context';
import { messages } from '../../messages';

const MAX_PRIMARY_SITES = 5;

type SiteWithAdvancedSearchURL = Site & {
  advancedSearchUrl: string;
};

export interface Props {
  sites: SiteWithAdvancedSearchURL[];
  linkComponent?: LinkComponent;
  onClick?: (href: string, e: React.MouseEvent) => any;
  isLoading: boolean;
}

const getActionSubjectIdForProduct = (product: Products) => {
  return `${product.toLowerCase()}AdvancedSearchLink`;
};

export const MultiSiteAdvancedSearchLink: FunctionComponent<
  Props & InjectedIntlProps
> = ({ sites, linkComponent, onClick, isLoading, intl }) => {
  const { fireAnalyticsEvent } = useAnalytics();

  const links = sites.map((site) => ({
    content: site.siteName,
    key: site.cloudId,
    href: site.advancedSearchUrl,
  }));

  const label = intl.formatMessage(messages.multi_site_advanced_search_label);

  const dropdownLabel = intl.formatMessage(
    messages.multi_site_advanced_search_more_dropdown,
    {
      count: links.length - MAX_PRIMARY_SITES,
    },
  );

  const onAdvancedSearchClicked = useCallback(
    ({ key }: { key: string }, e: React.MouseEvent<HTMLElement>) => {
      const site = sites.filter((s) => s.cloudId === key)[0];

      if (!site) {
        return;
      }

      fireAnalyticsEvent(
        onMultiSiteAdvancedSearchSelected({
          trigger: getTrigger(e),
          actionSubjectId: getActionSubjectIdForProduct(site.product),
          isLoading,
          newTab: isNewTab(e),
          destinationId: key,
          index: sites.map((s) => s.cloudId).indexOf(key),
        }),
      );
    },
    [fireAnalyticsEvent, isLoading, sites],
  );

  return (
    <SearchFooterLinks
      linkComponent={linkComponent}
      onClick={(link, e) => {
        onAdvancedSearchClicked(link, e);
        onClick && onClick(link.href, e);
      }}
      label={label}
      dropdownTriggerLabel={dropdownLabel}
      links={links}
      maxPrimaryLinks={MAX_PRIMARY_SITES}
    />
  );
};

export default injectIntl(MultiSiteAdvancedSearchLink);
