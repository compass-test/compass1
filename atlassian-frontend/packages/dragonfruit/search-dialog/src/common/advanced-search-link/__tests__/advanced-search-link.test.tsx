import React from 'react';
import { shallow, mount } from 'enzyme';
import { LinkComponent, LinkComponentProps } from '@atlassian/search-dialog';
import { Props, AdvancedSearchLink } from '../advanced-search-link';

describe('<AdvancedSearchLink />', () => {
  const createDummyComponent = (
    onMount: () => void = () => {},
  ): LinkComponent => {
    return class DummyComponent extends React.Component<LinkComponentProps> {
      componentDidMount() {
        onMount();
      }

      render() {
        return <div />;
      }
    };
  };

  const defaultProps: Props = {
    href: 'www.atlassian.com',
    isKeyboardHighlighted: false,
    onClick: () => {},
  };

  it('AdvancedSearchLink matches snapshot', () => {
    const wrapper = shallow(<AdvancedSearchLink {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('AdvancedSearchLink matches snapshot when keyboard highlighted', () => {
    const wrapper = shallow(
      <AdvancedSearchLink {...defaultProps} isKeyboardHighlighted />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('AdvancedSearchLink matches snapshot with custom component', () => {
    const wrapper = shallow(
      <AdvancedSearchLink
        {...defaultProps}
        linkComponent={createDummyComponent()}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('AdvancedSearchLink does not remount linkComponent when prop is updated but linkComponent is not', () => {
    const onMount = jest.fn();
    const DummyComponent = createDummyComponent(onMount);

    const wrapper = mount(
      <AdvancedSearchLink {...defaultProps} linkComponent={DummyComponent} />,
    );

    expect(onMount).toBeCalledTimes(1);

    wrapper.setProps({
      href: 'some/new/url',
      linkComponent: DummyComponent,
    });

    wrapper.update();

    expect(onMount).toBeCalledTimes(1);
  });

  it('AdvancedSearchLink does generate a new linkComponent when prop is updated and AdvancedSearchLink is updated', () => {
    const onMount = jest.fn();
    const DummyComponent = createDummyComponent(onMount);
    const NewDummyComponent = createDummyComponent(onMount);

    const wrapper = mount(
      <AdvancedSearchLink {...defaultProps} linkComponent={DummyComponent} />,
    );

    expect(onMount).toBeCalledTimes(1);

    wrapper.setProps({
      linkComponent: NewDummyComponent,
    });

    wrapper.update();

    expect(onMount).toBeCalledTimes(2);
  });
});
