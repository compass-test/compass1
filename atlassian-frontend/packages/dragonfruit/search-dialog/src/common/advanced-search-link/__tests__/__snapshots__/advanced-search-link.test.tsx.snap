// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedSearchLink /> AdvancedSearchLink matches snapshot 1`] = `
<Styled(Link)
  data-test-id="search-dialog-advanced-search-link"
  href="www.atlassian.com"
  isKeyboardHighlighted={false}
  onClick={[Function]}
/>
`;

exports[`<AdvancedSearchLink /> AdvancedSearchLink matches snapshot when keyboard highlighted 1`] = `
<Styled(Link)
  data-test-id="search-dialog-advanced-search-link"
  href="www.atlassian.com"
  isKeyboardHighlighted={true}
  onClick={[Function]}
/>
`;

exports[`<AdvancedSearchLink /> AdvancedSearchLink matches snapshot with custom component 1`] = `
<Styled(Link)
  data-test-id="search-dialog-advanced-search-link"
  href="www.atlassian.com"
  isKeyboardHighlighted={false}
  linkComponent={[Function]}
  onClick={[Function]}
/>
`;
